File: Module 6/61_Selection Sort/Cpp/Selection.cpp
#include <iostream>
using namespace std;

void swap(int &a, int &b) {
    int temp = a;
    a = b;
    b = temp;
}

void SelectionSort(int arr[], int n) {
    int min;

    for(int i = 0; i < n; i++) {
        min = i;

        for(int j = i + 1; j < n; j++) {
            if(arr[j] < arr[min]) {
                min = j;
            }
        }
        swap(arr[min], arr[i]);
    }
}

void displayArray(int arr[], int n) {
    for(int i = 0; i < n; i++) {
        cout << arr[i] << " ";
    }
}

int main() {
    int arr[] = {91, 32, 86, 37, 12};
    int n = sizeof(arr) / sizeof(arr[0]);

    cout << "Unsorted Array" << endl;
    displayArray(arr, n);
    cout << endl;

    SelectionSort(arr, n);
    cout << "Sorted Array" << endl;
    displayArray(arr, n);
    cout << endl;

    return 0;
}

File: Module 6/61_Selection Sort/Java/Selection.java
public class Selection {

    public static void swap(int[] arr, int i, int j) {
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }

    public static void selectionSort(int[] arr) {
        int n = arr.length;
        int min;

        for(int i = 0; i < n; i++) {
            min = i;

            for(int j = i + 1; j < n; j++) {
                if(arr[j] < arr[min]) {
                    min = j;
                }
            }
            swap(arr, min, i);
        }
    }

    public static void displayArray(int[] arr) {
        for(int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }
    }

    public static void main(String[] args) {
        int[] arr = {91, 32, 86, 37, 12};

        System.out.println("Unsorted Array");
        displayArray(arr);
        System.out.println();

        selectionSort(arr);
        System.out.println("Sorted Array");
        displayArray(arr);
    }
}
